include/xgboost/c_api.h

```diff
@@ -1221,7 +1221,7 @@ XGB_DLL int XGBoosterPredictFromCudaColumnar(BoosterHandle handle, char const *v
  * \brief Load model from existing file
  *
  * \param handle handle
- * \param fname File URI or file name.
+ * \param fname File URI or file name. The string must be UTF-8 encoded.
```

Comment:
Out of curiosity, what happens if it's not encoded in utf-8?

Reply:
If the path is ASCII, it's automatically a valid UTF-8 string.
If the path is not ASCII and is encoded with another encoding, you will get an undefined behavior.

That said, all current language bindings encode paths using UTF-8:

* Java: https://github.com/dmlc/xgboost/blob/7ce090e775e017f7850a279befddb1d694c0c9fe/jvm-packages/xgboost4j/src/native/xgboost4j.cpp#L690C30-L690C47
* Python: https://github.com/dmlc/xgboost/blob/7ce090e775e017f7850a279befddb1d694c0c9fe/python-package/xgboost/core.py#L2502 (`c_str` uses the UTF-8 encoding)
* R: I will need to file a follow-up PR to adopt UTF-8 encoding for `xgb.save` and `xgb.load` https://github.com/dmlc/xgboost/blob/7ce090e775e017f7850a279befddb1d694c0c9fe/R-package/R/xgb.save.R#L46